File: 88.merge-sorted-array.java
/*
 * @lc app=leetcode id=88 lang=java
 *
 * [88] Merge Sorted Array
 */

// @lc code=start
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int l = m - 1;
        int r = n - 1;

        while (l >= 0 && r >= 0) {
            if (nums1[l] > nums2[r]) {
                nums1[l + r + 1] = nums1[l];
                l--;
            } else {
                nums1[l + r + 1] = nums2[r];
                r--;
            }
        }
        
        for (int i = 0; i <= r; i++) {
            nums1[i] = nums2[i];
        }
    }
}
// @lc code=end

File: 88.merge-sorted-array.py
#
# @lc app=leetcode id=88 lang=python3
#
# [88] Merge Sorted Array
#

# @lc code=start
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        # # First Idea: Merge Sort
        # res = []
        # nums1[m:] = nums2
        
        # i = 0
        # j = m
        # while i <= m-1 and j <= m+n-1:
        #     if nums1[i] > nums1[j]:
        #         res.append(nums1[j])
        #         j += 1
        #     else:
        #         res.append(nums1[i])
        #         i += 1
        
        # if i <= m-1:
        #     res.extend(nums1[i:m])
        
        # if j <= m+n-1:
        #     res.extend(nums1[j:])
        
        # nums1[:] = res
        # Solution in discussion
        # nums1[m:] = nums2
        # nums1.sort()


        


# @lc code=end

